src/compiler/util.h

```diff
@@ -122,6 +122,30 @@ class PathBuilder {
     join(segment4);
   }
 
+  // Joins the given sub_path to this builder.
+  // The sub_path must use slashes as path separators.
+  void joinSlashPath(const std::string& sub_path) {
+    if (sub_path.empty()) return;
+    size_t last_start = 0;
```

Comment:
This seems pretty convoluted. Wouldn't it be easier to write with `sub_path.find_first_of`?

Reply:
Completely rewritten.